app/frontend/src/routes.ts

```diff
@@ -15,17 +15,39 @@ export const confirmChangeEmailRoute = "/confirm-email";
 
 export const signupRoute = "/signup";
 
-export const userRoute = "/user";
-export const aboutRoute = `${userRoute}/about`;
-export const homeRoute = `${userRoute}/home`;
-export const referencesRoute = `${userRoute}/references`;
-export const favoritesRoute = `${userRoute}/favorites`;
-export const photosRoute = `${userRoute}/photos`;
-
-export const editProfileRoute = `${userRoute}/edit`;
-export const editHostingPreferenceRoute = `${userRoute}/edit-hosting`;
-
-export const routeToUser = (username: string) => `${userRoute}/${username}`;
+// user
+
+const userBaseRoute = "/user";
+export type UserTab = "about" | "home" | "references" | "favorites" | "photos";
+export type EditUserTab = Extract<UserTab, "about" | "home">;
+
+export const userRoute = `${userBaseRoute}/:userId?/:tab?`;
```

Comment:
```suggestion
export const userRoute = `${userBaseRoute}/:username?/:tab?`;
```
Visiting other user profiles and tests are broken because of this line